api/types/app.go

```diff
@@ -234,6 +236,21 @@ func (a *AppV3) IsAWSConsole() bool {
 	return false
 }
 
+// IsDynamoDB returns true if this app is a DynamoDB app.
+func (a *AppV3) IsDynamoDB() bool {
+	if !a.IsAWSConsole() {
+		return false
+	}
+	// example uri that should match: https://console.aws.amazon.com/dynamdodbv2/home
+	// parts = ["https:", "", "console.aws.amazon.com", "dynamodbv2", "home"]
+	// relevant string is in the 4th part.
+	parts := strings.Split(a.Spec.URI, "/")
```

Comment:
What if a request originated from SDK or AWS CLI ? The URL will contain one for the[ DynamoDB endpoints  ](https://docs.aws.amazon.com/general/latest/gr/ddb.html) where `IsDynamoDB` function will be unable to detect the dynamoDB request.

Reply:
We only support AWS console dynamodb access I believe. See here:
https://github.com/gravitational/teleport/blob/ffc4f43f25df8c7f777b346372adaf1f72fef3d8/lib/srv/app/server.go#L668-L684

The non-console path in that code will use JWT. I don't know how to set that up for dynamodb and we have only docs for Elasticsearch. Our DynamoDB guide instructs users to use the console URI: https://goteleport.com/docs/application-access/guides/aws-console/#step-59-register-aws-console-application-in-teleport

Reply:
Also, this is checking the Teleport app spec, not the request. It does not matter where the requests come from - the app is configured with a console URI.

Reply:
Changed this to use the signing name for the request: "dynamodb" and "dax". Updated test to use a generic console app uri instead of a dynamodb console uri to verify this